ci/osx-build-config/extensions_build_config.bzl

```diff
@@ -16,7 +16,6 @@ EXTENSIONS = {
     "envoy.network.dns_resolver.getaddrinfo": "//source/extensions/network/dns_resolver/getaddrinfo:config",
     "envoy.transport_sockets.http_11_proxy": "//source/extensions/transport_sockets/http_11_proxy:upstream_config",
     "envoy.transport_sockets.raw_buffer": "//source/extensions/transport_sockets/raw_buffer:config",
-    "envoy.transport_sockets.tls": "//source/extensions/transport_sockets/tls:config",
```

Comment:
Following the pattern of other core/built-in things used at extension points, such as the tcp_proxy network filter and the raw_buffer transport socket, I think there should still be a stub-extension that can be enabled/disabled here, and just the factory code live in `source/extensions`